### PR TITLE
Set up `enable_mem_deduplication` and `mem_pool_prune_interval` in co…

### PR DESCRIPTION
**Pull request by rafal-ch**
This PR adds the following parameters to the configs that require them and also sets the values as follows:
```
enable_mem_deduplication = true
mem_pool_prune_interval = 4096
```

### CODE REVIEW

File: resources/local/config.toml
Comment (afck) [inline resources/local/config.toml:251]: Would be good to clarify that "5" means "every 5th time something is put in the pool the cache is swept". (@marc-casperlabs: Please confirm that's correct!)

File: utils/nctl/sh/scenarios/configs/bond_its.config.toml
Comment (afck) [inline utils/nctl/sh/scenarios/configs/bond_its.config.toml:251]: Or was it intentionally lower in the tests? (Maybe because we're running more nodes, which therefore already use up more memory?)

**Review by afck — APPROVED** [review]
Looks good to me, but please also wait for @marc-casperlabs's approval.

**Review by Fraser999 — APPROVED** [review]

File: resources/local/config.toml
Comment (marc-casperlabs) [reply resources/local/config.toml:251]: Yes, that's correct! :)

File: utils/nctl/sh/scenarios/configs/bond_its.config.toml
Comment (marc-casperlabs) [reply utils/nctl/sh/scenarios/configs/bond_its.config.toml:251]: I just don't know what a good value is yet. There's three things that this number mainly affects: Latency, CPU usage and memory usage.

Latency because higher numbers mean that the potential pause for pruning is longer the more items can be accumulated, so lower numbers mean less latency. On the other hand, if set too low, the CPU usage becomes closer to `O(n^2)`, which would be checking on every single request. Finally lower numbers are obviously better for memory usage.

To be honest, I think with these numbers, we can discount the memory usage as a non-issue - it is 8 or 16 bytes per entry currently, so 4096 is 64 KB of maximum memory usage. Latency wise, iterating over 4096 entries is also next to nothing, I think, and I believe the CPU time to be negligble as well. We could possibly go higher, but if 4096 checks the boxes, why not use it?

As for why this was at 1024 before, I believe it was lowered in tests mainly to have it triggered faster - remember, you have to request 4096 things first for the code to be run the first time :)

**Pull request merged**
